kotlin/src/main/kotlin/io/logto/sdk/core/util/UriUtils.kt

```diff
@@ -0,0 +1,36 @@
+package io.logto.sdk.core.util
+
+import io.logto.sdk.core.constant.QueryKey
+import io.logto.sdk.core.exception.LogtoException
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+
+object UriUtils {
+    fun verifyAndParseCodeFromCallbackUri(
+        callbackUri: String,
+        redirectUri: String,
+        state: String,
+    ): String {
+        if (!callbackUri.startsWith(redirectUri)) {
+            throw LogtoException.VerificationException(LogtoException.Verification.URI_MISMATCHED)
+        }
+
+        val parsedUri = callbackUri.toHttpUrlOrNull()
```

Comment:
not sure, but, if we have passed `callbackURI` start with `redirectUri` validation above, could it still be an invalid Uri here?

Reply:
refer to the [slack thread](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1642580813321000)

Reply:
updated